Automattic/wp-calypso
# Registrationless checkout: Add email field validation

### Changes proposed in this Pull Request

**This is one among a set of stacked PRs that implements userless checkout A/B test. For more reading on the userless
checkout project, read pau2Xa-1hW-p2.**

This PR implements the following:
* Connects the email field to the validation endpoint that verifies if the entered email address is valid.
* Displays an error message below the email field in case of an error. 

### NOTE:
- Depends on D45659-code.
- This PR is a partial implementation only. You will not be able to process the transaction yet, subsequent PRs handle this.

### Testing instructions

#### NOTE: Depends on D45659-code, so please apply this first.

**Scenario 1 - Test invalid email formats**

* Navigate to http://calypso.localhost:3000/, set your locale/geo to EN-US, and assign yourself to _variantUserless_ of `userlessCheckout` test and _control_ of `newSiteGutenbergOnboarding` test.
* Go to /start, select a paid plan and proceed to checkout. To test email validation, you will need to type something in the email field and click "Continue".
* In the email field, try invalid email formats and verify that you are shown an error - e.g `rand@something`, `rand#gmail.com` etc.
* Leave the email field blank and verify you get an error.
* Repeat above steps with a domain in cart - email validation should work in both Billing Info section and Contact Info section(check screenshots in https://github.com/Automattic/wp-calypso/pull/44211).

**Scenario 2 - Test email taken message**
* Navigate to http://calypso.localhost:3000/, set your locale/geo to EN-US, and assign yourself to _variantUserless_ of `userlessCheckout` test and _control_ of `newSiteGutenbergOnboarding` test.
* Go to /start, select a paid plan and proceed to checkout.
* Type an email ID that is tied to an existing wpcom account. You should get a message. You will see an error message - "That email address is already in use. If you have an existing account, please log in.".
* Verify both in the Billing Info and Contact Info sections (i.e without and with domain in cart).
* _Note - Clicking the login link will not work as expected yet. This will be fixed in https://github.com/Automattic/wp-calypso/pull/44238_
